### ISSUE TITLE
Update Checkstyle API

### ISSUE THREAD

**Issue by svilenvul** [openhab/static-code-analysis#225]
Currently we are using 8.1, while 8.5. is released.

Looking at the [release notes](http://checkstyle.sourceforge.net/releasenotes.html#Release_8.5), I see some braking API changes that will affect us... :
- Disallowing user to use incomplete fully qualified Check names in config file;
- BaseCheckTestSupport is deprecated;

Most of the changes are bug fixes, I don`t see many new checks (new Check: single line annotation location), so maybe we can wait a bit with the update.

**Comment by svilenvul** [comment]
@kssimova mentioned that some false positives in the checks for javadoc are fixed in the newer checkstyle realaeses. Could you provide more details on that ?

**Comment by kssimova** [comment]
**UnusedImportsCheck** reports imports as unused, even though they are used in a Javadoc. For example: 
```java
import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;

/**
 * The normalizer for the {@link ConfigDescriptionParameter.Type#TEXT}. It basically ensures that the given value will
 * turned into its {@link String} representation.
 *
 * @author Simon Kaufmann - initial contribution and API.
 * @author Thomas Höfer - made class final and minor javadoc changes
 */
final class TextNormalizer extends AbstractNormalizer {

    @Override
    public Object doNormalize(Object value) {
        if (value == null) {
            return value;
        }
        if (value instanceof String) {
            return value;
        }
        return value.toString();
    }

}

```
I did some research and found that this problem is fixed in **Checkstyle** 8.3 : https://github.com/checkstyle/checkstyle/issues/3453, while we're currently using 8.1.

**Comment by svilenvul** [comment]
Definitely not the single issue being fixed, but the first one that observed in the reports for ESH/openHAB.
